icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -118,6 +124,15 @@ public void takeReciprocal() {
         return result;
     }
 
+    public List<MeasureUnitImpl> extractIndividualUnitsWithIndex() {
```

Comment:
C++: Better than std::pair would be to create your own struct that contains the index and the MeasureUnitImpl.  The struct should inherit UMemory.  You can't use std::pair directly in a MaybeStackVector because it would need to inherit UMemory.

Reply:
We now have a struct. I'm marking this as resolved.